charts/redpanda/templates/tests/test-kafka-nodelete.yaml

```diff
@@ -89,12 +90,14 @@ spec:
          
           # now check if we can delete the topic (we should not)
           rpk topic delete my_sample_topic
+
+          {{- if has "my_sample_topic" $noDeleteTopics }}
```

Comment:
I was thinking that check would be at the top, so the Pod would not be executed at all.